src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -545,11 +552,13 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
 			case IntrinsicId.Nullable_GetUnderlyingType:
 				var values = new List<SingleValue> ();
 				foreach (var singlevalue in argumentValues[0].AsEnumerable ()) {
-					if (singlevalue is ValueWithDynamicallyAccessedMembers damValue) {
-						values.Add (GetMethodReturnValue (calledMethod, damValue.DynamicallyAccessedMemberTypes));
-					} else if (singlevalue is NullableSystemTypeValue systemNullableTypeValue) {
-						values.AddRange (systemNullableTypeValue.UnderlyingTypeValue);
-					}
+					values.Add (singlevalue switch {
+						SystemTypeValue systemType => systemType,
```

Comment:
Maybe add an assert (if possible) that the type is not Nullable<T> - since below we rely on that.